flow/connectors/kafka/kafka.go

```diff
@@ -0,0 +1,330 @@
+	case "Manual":
+		optionalOpts = append(optionalOpts, kgo.RecordPartitioner(kgo.ManualPartitioner()))
+	case "RoundRobin":
+		optionalOpts = append(optionalOpts, kgo.RecordPartitioner(kgo.RoundRobinPartitioner()))
+	case "StickyKey":
+		optionalOpts = append(optionalOpts, kgo.RecordPartitioner(kgo.StickyKeyPartitioner(nil)))
+	case "Sticky":
+		optionalOpts = append(optionalOpts, kgo.RecordPartitioner(kgo.StickyPartitioner()))
+	}
+	if config.Username != "" {
+		switch config.Sasl {
+		case "PLAIN":
+			auth := plain.Auth{User: config.Username, Pass: config.Password}
+			optionalOpts = append(optionalOpts, kgo.SASL(auth.AsMechanism()))
+		case "SCRAM-SHA-256":
+			auth := scram.Auth{User: config.Username, Pass: config.Password}
+			optionalOpts = append(optionalOpts, kgo.SASL(auth.AsSha256Mechanism()))
+		case "SCRAM-SHA-512":
+			auth := scram.Auth{User: config.Username, Pass: config.Password}
+			optionalOpts = append(optionalOpts, kgo.SASL(auth.AsSha512Mechanism()))
+		default:
+			return nil, fmt.Errorf("unsupported SASL mechanism: %s", config.Sasl)
+		}
+	}
+	client, err := kgo.NewClient(optionalOpts...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kafka client: %w", err)
+	}
+
+	pgMetadata, err := metadataStore.NewPostgresMetadataStore(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &KafkaConnector{
+		client:     client,
+		pgMetadata: pgMetadata,
+		logger:     logger.LoggerFromCtx(ctx),
+	}, nil
+}
+
+func (c *KafkaConnector) Close() error {
+	if c != nil {
+		c.client.Close()
+	}
+	return nil
+}
+
+func (c *KafkaConnector) ConnectionActive(ctx context.Context) error {
+	return c.client.Ping(ctx)
+}
+
+func (c *KafkaConnector) CreateRawTable(ctx context.Context, req *protos.CreateRawTableInput) (*protos.CreateRawTableOutput, error) {
+	return &protos.CreateRawTableOutput{TableIdentifier: "n/a"}, nil
+}
+
+func (c *KafkaConnector) GetLastSyncBatchID(ctx context.Context, jobName string) (int64, error) {
+	return c.pgMetadata.GetLastBatchID(ctx, jobName)
+}
+
+func (c *KafkaConnector) GetLastOffset(ctx context.Context, jobName string) (int64, error) {
+	return c.pgMetadata.FetchLastOffset(ctx, jobName)
+}
+
+func (c *KafkaConnector) SetLastOffset(ctx context.Context, jobName string, offset int64) error {
+	return c.pgMetadata.UpdateLastOffset(ctx, jobName, offset)
+}
+
+func (c *KafkaConnector) NeedsSetupMetadataTables(_ context.Context) bool {
+	return false
+}
+
+func (c *KafkaConnector) SetupMetadataTables(_ context.Context) error {
+	return nil
+}
+
+func (c *KafkaConnector) ReplayTableSchemaDeltas(_ context.Context, flowJobName string, schemaDeltas []*protos.TableSchemaDelta) error {
+	return nil
+}
+
+func (c *KafkaConnector) SyncFlowCleanup(ctx context.Context, jobName string) error {
+	return c.pgMetadata.DropMetadata(ctx, jobName)
+}
+
+func (c *KafkaConnector) SyncRecords(ctx context.Context, req *model.SyncRecordsRequest) (*model.SyncResponse, error) {
```

Comment:
can we refactor this to smaller methods?

Reply:
Split out methods for `loadScript`/`lvalueToKafkaRecord`